mt-kahypar/io/command_line_options.cpp

```diff
@@ -486,25 +486,48 @@ namespace mt_kahypar {
-             po::value<bool>((!initial_partitioning ? &context.refinement.global_fm.refine_until_no_improvement :
-                              &context.initial_partitioning.refinement.global_fm.refine_until_no_improvement))->value_name(
+             po::value<bool>((!initial_partitioning ? &context.refinement.global.refine_until_no_improvement :
+                              &context.initial_partitioning.refinement.global.refine_until_no_improvement))->value_name(
                      "<bool>")->default_value(false),
              "Executes a globalized FM local search as long as it finds an improvement on the current partition.")
+            ((initial_partitioning ? "i-r-global-fm-type" : "r-global-fm-type"),
+             po::value<std::string>()->value_name("<string>")->notifier(
+                     [&, initial_partitioning](const std::string& type) {
+                       if (initial_partitioning) {
+                         context.initial_partitioning.refinement.global.fm_algorithm = fmAlgorithmFromString(type);
+                       } else {
+                         context.refinement.global.fm_algorithm = fmAlgorithmFromString(type);
+                       }
+                     })->default_value("kway_fm"),
+             "FM Algorithm for the globalized FM local search:\n"
+             "- kway_fm\n"
+             "- unconstrained_fm\n"
+             "- do_nothing")
             ((initial_partitioning ? "i-r-global-fm-seed-nodes" : "r-global-fm-seed-nodes"),
-             po::value<size_t>((initial_partitioning ? &context.initial_partitioning.refinement.global_fm.num_seed_nodes :
-                                &context.refinement.global_fm.num_seed_nodes))->value_name("<size_t>")->default_value(25),
+             po::value<size_t>((initial_partitioning ? &context.initial_partitioning.refinement.global.num_seed_nodes :
+                                &context.refinement.global.num_seed_nodes))->value_name("<size_t>")->default_value(25),
              "Number of nodes to start the 'highly localized FM' with during the globalized FM local search.")
             ((initial_partitioning ? "i-r-global-fm-obey-minimal-parallelism" : "r-global-fm-obey-minimal-parallelism"),
              po::value<bool>(
-                     (initial_partitioning ? &context.initial_partitioning.refinement.global_fm.obey_minimal_parallelism :
-                      &context.refinement.global_fm.obey_minimal_parallelism))->value_name("<bool>")->default_value(true),
+                     (initial_partitioning ? &context.initial_partitioning.refinement.global.obey_minimal_parallelism :
+                      &context.refinement.global.obey_minimal_parallelism))->value_name("<bool>")->default_value(true),
              "If true, then the globalized FM local search stops if more than a certain number of threads are finished.")
+            ((initial_partitioning ? "i-r-global-use-lp" : "r-global-use-lp"),
```

Comment:
Is there a reason to have a different interface for global LP than for global FM? For FM it's {unconstrained, constrained, do_nothing}. For LP it's {use LP, do not use LP} x {unconstrained, constrained}.

Reply:
The reason was that we don't need two different LP refiner instances when doing it in this way. However, it is indeed better to provide a more uniform interface :+1: